## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374764`

### Reviewed points
<details>
  <summary>:white_check_mark: 23 | :red_circle: 0</summary>

- [x] package path fits guidelines (e.g., `pkgs/by-name/so/some-package/package.nix` for top-level packages).
- [x] Package name fits guidelines.
- [x] Package version fits guidelines.
- [x] Source is fetched using the appropriate function.
- [x] `tag` is used instead of `rev` for tag versions.
- [x] `hash` is used instead of `sha256` for hashes.

- [x] Build time only dependencies are declared in `nativeBuildInputs` or its equivalent in the used builder.
- [x] The list of `phases` is not overridden.
- [x] When a phase (like `installPhase`) is overridden, it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] Patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed.
- [x] Patches that are remotely available are fetched rather than vendored.
- [x] Package builds on x86_64-linux.
- [x] Executables tested on x86_64-linux.

- [x] `meta.description` is set and fits guidelines.
	- [x] Descriptions should not contain complete sentences and should consist of phrases that fill the blank in " is a/an/the ____" (starting with a capital letter).
- [x] `meta.license` fits upstream license.
- [x] `meta.platforms` is set.
- [x] `meta.maintainers` is set.
- [x] `meta.mainProgram` is set, if applicable.

- [x] Whatever is needed is brought into scope instead of using top-level pkgs to make it easier to override.
- [x] General `with lib;` is not used.
- [x] If possible, update script is provided.
- [x] If possible, tests are provided.

</details>

### Possible improvements

### Comments

That being said, the game immediately crashed on me when trying to create a character during rerolling some character traits or something.

```
Traceback (most recent call last):
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/bin/.pqcli-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/__main__.py", line 56, in main
    ui.run()
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/ui/curses/__init__.py", line 95, in run
    self._view.keypress(key)
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/ui/curses/views/create_character_view/choose_character_stats_view.py", line 58, in keypress
    self.on_confirm(self._stats)
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/ui/curses/event_handler.py", line 10, in __call__
    callback(*args, **kwargs)
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/ui/curses/__init__.py", line 145, in <lambda>
    view.on_confirm += lambda stats: self._create_character(
                                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/ui/curses/__init__.py", line 185, in _create_character
    player = create_player(name, race, class_, stats)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/mechanic.py", line 929, in create_player
    random.seed(now)
  File "/nix/store/i2nqm97vgf0621821pn5bl7x4k4zz1pg-pq-cli-1.0.2/lib/python3.12/site-packages/pqcli/random.py", line 6, in seed
    random.seed(source)
  File "/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/lib/python3.12/random.py", line 167, in seed
    raise TypeError('The only supported seed types are: None,\n'
TypeError: The only supported seed types are: None,
int, float, str, bytes, and bytearray.
```

--- a/pkgs/by-name/pq/pq-cli/package.nix
+++ b/pkgs/by-name/pq/pq-cli/package.nix
@@ -1,0 +1,46 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+  unstableGitUpdater,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "pq-cli";
+  version = "1.0.2-unstable-2024-07-15";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "rr-";
+    repo = "pq-cli";
+    rev = "4122e936c87436552f35ff7972d6e543ba6c837b";
+    hash = "sha256-PvHkTjPjOubhl9gfPTF7yNaFgg2GLk8y+PtF09fpaII=";
+  };
+
+  build-system = with python3Packages; [
+    setuptools
+    poetry-core
+  ];
+
+  dependencies = with python3Packages; [
+    xdg
+    xdg-base-dirs
+    urwid
+    urwid-readline
+  ];
+
+  pythonRelaxDeps = [
+    "urwid-readline"
+  ];
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "Progress Quest: the CLI edition";
+    homepage = "https://github.com/rr-/pq-cli";
+    changelog = "https://github.com/rr-/pq-cli/releases/tag/${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ genga898 ];
+    mainProgram = "pqcli";
+  };
+}

